test/test_cuda.py

```diff
@@ -903,6 +908,25 @@ def test_type_conversions(self):
         self.assertIsInstance(y.cuda().float().cpu(), torch.FloatStorage)
         self.assertIsInstance(y.cuda().float().cpu().int(), torch.IntStorage)
 
+    def test_mul_intertype_scalar(self):
+        x = torch.tensor(1.5, device='cuda')
+        y = torch.tensor(3, dtype=torch.int32, device='cuda')
+
+        self.assertEqual(x * y, 4.5)
+        self.assertEqual(y * x, 4.5)
+        with self.assertRaisesRegex(RuntimeError, 'expected type'):
+            y *= x
+        x *= y
+        self.assertEqual(x, 4.5)
+
+        x = torch.tensor(1.5, device='cuda', dtype=torch.float16)
+        self.assertEqual(x * y, 4.5)
+        # half * int currently promotes to double
```

Comment:
change this?

Reply:
I'll make that change in a subsequent PR